### ISSUE TITLE
Unwanted join notifications from one Discord server to another

### ISSUE THREAD

**Issue by halian** [42wim/matterbridge#1612]
**Describe the bug**
My MatterBridge instance has two Discord servers, each bridged to mutually exclusive sets of IRC channels. When someone joins Discord B, a notification is posted by the bot to Discord A.

**To Reproduce**
Unknown. Nothing obviously leading to the bug can be found in the configuration file.

**Expected behavior**
Notifications to be posted to Discord A only when someone joins IRC A, and same for Discord B/IRC B.

**Screenshots/debug logs**
```
[0043] DEBUG discord:      [memberAdd:bridge/discord/handlers.go:183] <= Sending message from discord.safiralliance to gateway
[0043] DEBUG discord:      [memberAdd:bridge/discord/handlers.go:184] <= Message is config.Message{Text:"Ketsuban joins", Channel:"", Username:"system", UserID:"", Avatar:"", Account:"discord.safiralliance", Event:"join_leave", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [memberAdd:bridge/discord/handlers.go:183] <= Sending message from discord.alden-ic to gateway
[0043] DEBUG discord:      [memberAdd:bridge/discord/handlers.go:184] <= Message is config.Message{Text:"Ketsuban joins", Channel:"", Username:"system", UserID:"", Avatar:"", Account:"discord.alden-ic", Event:"join_leave", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [memberAdd:bridge/discord/handlers.go:183] <= Sending message from discord.alden-ooc to gateway
[0043] DEBUG discord:      [memberAdd:bridge/discord/handlers.go:184] <= Message is config.Message{Text:"Ketsuban joins", Channel:"", Username:"system", UserID:"", Avatar:"", Account:"discord.alden-ooc", Event:"join_leave", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [memberUpdate:bridge/discord/handlers.go:149] discord.safiralliance: memberupdate: user Ketsuban (nick ) changes nick to
[0043] DEBUG gateway:      [SendMessage:gateway/gateway.go:479] => Sending config.Message{Text:"Ketsuban joins", Channel:"", Username:"system", UserID:"", Avatar:"", Account:"discord.alden-ooc", Event:"join_leave", Protocol:"discord", Gateway:"aldenooc", ParentID:"", Timestamp:time.Time{wall:0xc052f9d22ea3f64b, ext:43697216555, loc:(*time.Location)(0x339b760)}, ID:"", Extra:map[string][]interface {}(nil)} from discord.alden-ooc () to discord.alden-ooc (ID:REDACTED)
[0043] DEBUG discord:      [Send:bridge/discord/discord.go:235] => Receiving config.Message{Text:"Ketsuban joins", Channel:"ID:REDACTED", Username:"<s\u200bystem> ", UserID:"", Avatar:"", Account:"discord.alden-ooc", Event:"join_leave", Protocol:"discord", Gateway:"aldenooc", ParentID:"", Timestamp:time.Time{wall:0xc052f9d22ea3f64b, ext:43697216555, loc:(*time.Location)(0x339b760)}, ID:"", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [handleEventWebhook:bridge/discord/webhook.go:139] Processing webhook sending for message &config.Message{Text:"Ketsuban joins", Channel:"ID:REDACTED", Username:"<s\u200bystem> ", UserID:"", Avatar:"", Account:"discord.alden-ooc", Event:"join_leave", Protocol:"discord", Gateway:"aldenooc", ParentID:"", Timestamp:time.Time{wall:0xc052f9d22ea3f64b, ext:43697216555, loc:(*time.Location)(0x339b760)}, ID:"", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:86] == Receiving event &discordgo.Message{ID:"REDACTED", ChannelID:"REDACTED", GuildID:"REDACTED", Content:"Ketsuban joins", Timestamp:"2021-10-17T00:35:20.856000+00:00", EditedTimestamp:"", MentionRoles:[]string{}, TTS:false, MentionEveryone:false, Author:(*discordgo.User)(0xc0006a17c0), Attachments:[]*discordgo.MessageAttachment{}, Embeds:[]*discordgo.MessageEmbed{}, Mentions:[]*discordgo.User{}, Reactions:[]*discordgo.MessageReactions(nil), Pinned:false, Type:0, WebhookID:"REDACTED", Member:(*discordgo.Member)(nil), MentionChannels:[]*discordgo.Channel(nil), Activity:(*discordgo.MessageActivity)(nil), Application:(*discordgo.MessageApplication)(nil), MessageReference:(*discordgo.MessageReference)(nil), Flags:0}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:135] <= Sending message from <system> on discord.safiralliance to gateway
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:136] <= Message is config.Message{Text:"Ketsuban joins", Channel:"", Username:"<system>", UserID:"REDACTED", Avatar:"https://cdn.discordapp.com/avatars/REDACTED/5619d994e4cc51a5199fa6ad38e09fe3.jpg", Account:"discord.safiralliance", Event:"", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"REDACTED", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:86] == Receiving event &discordgo.Message{ID:"REDACTED", ChannelID:"REDACTED", GuildID:"REDACTED", Content:"Ketsuban joins", Timestamp:"2021-10-17T00:35:20.856000+00:00", EditedTimestamp:"", MentionRoles:[]string{}, TTS:false, MentionEveryone:false, Author:(*discordgo.User)(0xc0006a1860), Attachments:[]*discordgo.MessageAttachment{}, Embeds:[]*discordgo.MessageEmbed{}, Mentions:[]*discordgo.User{}, Reactions:[]*discordgo.MessageReactions(nil), Pinned:false, Type:0, WebhookID:"REDACTED", Member:(*discordgo.Member)(nil), MentionChannels:[]*discordgo.Channel(nil), Activity:(*discordgo.MessageActivity)(nil), Application:(*discordgo.MessageApplication)(nil), MessageReference:(*discordgo.MessageReference)(nil), Flags:0}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:135] <= Sending message from <system> on discord.alden-ic to gateway
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:136] <= Message is config.Message{Text:"Ketsuban joins", Channel:"aldenooc", Username:"<system>", UserID:"REDACTED", Avatar:"https://cdn.discordapp.com/avatars/REDACTED/5619d994e4cc51a5199fa6ad38e09fe3.jpg", Account:"discord.alden-ic", Event:"", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"REDACTED", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:86] == Receiving event &discordgo.Message{ID:"REDACTED", ChannelID:"REDACTED", GuildID:"REDACTED", Content:"<:happy_safir:> Ārislānđa, <@Ketsuban>, and welcome to the Safir Alliance! <:wave_safir:> Please read the <#rules>, then head over to https://roleypoly.com to set up your roles.", Timestamp:"2021-10-17T00:35:20.879000+00:00", EditedTimestamp:"", MentionRoles:[]string{}, TTS:false, MentionEveryone:false, Author:(*discordgo.User)(0xc0009e2140), Attachments:[]*discordgo.MessageAttachment{}, Embeds:[]*discordgo.MessageEmbed{}, Mentions:[]*discordgo.User{(*discordgo.User)(0xc0009e20a0)}, Reactions:[]*discordgo.MessageReactions(nil), Pinned:false, Type:0, WebhookID:"", Member:(*discordgo.Member)(0xc00013c380), MentionChannels:[]*discordgo.Channel(nil), Activity:(*discordgo.MessageActivity)(nil), Application:(*discordgo.MessageApplication)(nil), MessageReference:(*discordgo.MessageReference)(nil), Flags:0}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:86] == Receiving event &discordgo.Message{ID:"REDACTED", ChannelID:"REDACTED", GuildID:"REDACTED", Content:"<:happy_safir:> Ārislānđa, <@Ketsuban>, and welcome to the Safir Alliance! <:wave_safir:> Please read the <#rules>, then head over to https://roleypoly.com to set up your roles.", Timestamp:"2021-10-17T00:35:20.879000+00:00", EditedTimestamp:"", MentionRoles:[]string{}, TTS:false, MentionEveryone:false, Author:(*discordgo.User)(0xc00086aa00), Attachments:[]*discordgo.MessageAttachment{}, Embeds:[]*discordgo.MessageEmbed{}, Mentions:[]*discordgo.User{(*discordgo.User)(0xc00086a960)}, Reactions:[]*discordgo.MessageReactions(nil), Pinned:false, Type:0, WebhookID:"", Member:(*discordgo.Member)(0xc0001f8700), MentionChannels:[]*discordgo.Channel(nil), Activity:(*discordgo.MessageActivity)(nil), Application:(*discordgo.MessageApplication)(nil), MessageReference:(*discordgo.MessageReference)(nil), Flags:0}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:86] == Receiving event &discordgo.Message{ID:"REDACTED", ChannelID:"REDACTED", GuildID:"REDACTED", Content:"<:happy_safir:> Ārislānđa, <@Ketsuban>, and welcome to the Safir Alliance! <:wave_safir:> Please read the <#rules>, then head over to https://roleypoly.com to set up your roles.", Timestamp:"2021-10-17T00:35:20.879000+00:00", EditedTimestamp:"", MentionRoles:[]string{}, TTS:false, MentionEveryone:false, Author:(*discordgo.User)(0xc00086ab40), Attachments:[]*discordgo.MessageAttachment{}, Embeds:[]*discordgo.MessageEmbed{}, Mentions:[]*discordgo.User{(*discordgo.User)(0xc00086aaa0)}, Reactions:[]*discordgo.MessageReactions(nil), Pinned:false, Type:0, WebhookID:"", Member:(*discordgo.Member)(0xc0001f87e0), MentionChannels:[]*discordgo.Channel(nil), Activity:(*discordgo.MessageActivity)(nil), Application:(*discordgo.MessageApplication)(nil), MessageReference:(*discordgo.MessageReference)(nil), Flags:0}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:135] <= Sending message from Tatsu on discord.safiralliance to gateway
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:136] <= Message is config.Message{Text:":happy_safir: Ārislānđa, <@Ketsuban>, and welcome to the Safir Alliance! :wave_safir: Please read the #rules, then head over to https://roleypoly.com to set up your roles.", Channel:"general", Username:"Tatsumaki", UserID:"REDACTED", Avatar:"https://cdn.discordapp.com/avatars/172002275412279296/43fc1879641f6239d6e4038eaa7dabbc.jpg", Account:"discord.safiralliance", Event:"", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"REDACTED", Extra:map[string][]interface {}(nil)}
[0043] DEBUG gateway:      [SendMessage:gateway/gateway.go:498] mID discord.alden-ooc: REDACTED
[0043] DEBUG gateway:      [func1:gateway/gateway.go:488] => Send from discord.alden-ooc () to discord.alden-ooc (ID:REDACTED) took 140.586817ms
[0043] DEBUG gateway:      [SendMessage:gateway/gateway.go:479] => Sending config.Message{Text:"Ketsuban joins", Channel:"", Username:"system", UserID:"", Avatar:"", Account:"discord.alden-ooc", Event:"join_leave", Protocol:"discord", Gateway:"chorrusooc", ParentID:"", Timestamp:time.Time{wall:0xc052f9d2370cc910, ext:43838304014, loc:(*time.Location)(0x339b760)}, ID:"", Extra:map[string][]interface {}(nil)} from discord.alden-ooc () to discord.alden-ooc (ID:REDACTED)
[0043] DEBUG discord:      [Send:bridge/discord/discord.go:235] => Receiving config.Message{Text:"Ketsuban joins", Channel:"ID:REDACTED", Username:"<s\u200bystem> ", UserID:"", Avatar:"", Account:"discord.alden-ooc", Event:"join_leave", Protocol:"discord", Gateway:"chorrusooc", ParentID:"", Timestamp:time.Time{wall:0xc052f9d2370cc910, ext:43838304014, loc:(*time.Location)(0x339b760)}, ID:"", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [handleEventWebhook:bridge/discord/webhook.go:139] Processing webhook sending for message &config.Message{Text:"Ketsuban joins", Channel:"ID:REDACTED", Username:"<s\u200bystem> ", UserID:"", Avatar:"", Account:"discord.alden-ooc", Event:"join_leave", Protocol:"discord", Gateway:"chorrusooc", ParentID:"", Timestamp:time.Time{wall:0xc052f9d2370cc910, ext:43838304014, loc:(*time.Location)(0x339b760)}, ID:"", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:135] <= Sending message from Tatsu on discord.alden-ic to gateway
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:136] <= Message is config.Message{Text:":happy_safir: Ārislānđa, <@Ketsuban>, and welcome to the Safir Alliance! :wave_safir: Please read the #, then head over to https://roleypoly.com to set up your roles.", Channel:"", Username:"Tatsumaki", UserID:"REDACTED", Avatar:"https://cdn.discordapp.com/avatars/172002275412279296/43fc1879641f6239d6e4038eaa7dabbc.jpg", Account:"discord.alden-ic", Event:"", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"REDACTED", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:135] <= Sending message from Tatsu on discord.alden-ooc to gateway
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:136] <= Message is config.Message{Text:":happy_safir: Ārislānđa, <@Ketsuban>, and welcome to the Safir Alliance! :wave_safir: Please read the #, then head over to https://roleypoly.com to set up your roles.", Channel:"", Username:"Tatsumaki", UserID:"REDACTED", Avatar:"https://cdn.discordapp.com/avatars/172002275412279296/43fc1879641f6239d6e4038eaa7dabbc.jpg", Account:"discord.alden-ooc", Event:"", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"REDACTED", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:86] == Receiving event &discordgo.Message{ID:"REDACTED", ChannelID:"REDACTED", GuildID:"REDACTED", Content:"Ketsuban joins", Timestamp:"2021-10-17T00:35:20.974000+00:00", EditedTimestamp:"", MentionRoles:[]string{}, TTS:false, MentionEveryone:false, Author:(*discordgo.User)(0xc0009e2460), Attachments:[]*discordgo.MessageAttachment{}, Embeds:[]*discordgo.MessageEmbed{}, Mentions:[]*discordgo.User{}, Reactions:[]*discordgo.MessageReactions(nil), Pinned:false, Type:0, WebhookID:"REDACTED", Member:(*discordgo.Member)(nil), MentionChannels:[]*discordgo.Channel(nil), Activity:(*discordgo.MessageActivity)(nil), Application:(*discordgo.MessageApplication)(nil), MessageReference:(*discordgo.MessageReference)(nil), Flags:0}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:135] <= Sending message from <system> on discord.safiralliance to gateway
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:136] <= Message is config.Message{Text:"Ketsuban joins", Channel:"", Username:"<system>", UserID:"REDACTED", Avatar:"https://cdn.discordapp.com/avatars/REDACTED/a8cc07b5a83c3755d3c234bdd7fb5c0f.jpg", Account:"discord.safiralliance", Event:"", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"REDACTED", Extra:map[string][]interface {}(nil)}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:86] == Receiving event &discordgo.Message{ID:"REDACTED", ChannelID:"REDACTED", GuildID:"REDACTED", Content:"Ketsuban joins", Timestamp:"2021-10-17T00:35:20.974000+00:00", EditedTimestamp:"", MentionRoles:[]string{}, TTS:false, MentionEveryone:false, Author:(*discordgo.User)(0xc00086afa0), Attachments:[]*discordgo.MessageAttachment{}, Embeds:[]*discordgo.MessageEmbed{}, Mentions:[]*discordgo.User{}, Reactions:[]*discordgo.MessageReactions(nil), Pinned:false, Type:0, WebhookID:"REDACTED", Member:(*discordgo.Member)(nil), MentionChannels:[]*discordgo.Channel(nil), Activity:(*discordgo.MessageActivity)(nil), Application:(*discordgo.MessageApplication)(nil), MessageReference:(*discordgo.MessageReference)(nil), Flags:0}
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:135] <= Sending message from <system> on discord.alden-ic to gateway
[0043] DEBUG discord:      [messageCreate:bridge/discord/handlers.go:136] <= Message is config.Message{Text:"Ketsuban joins", Channel:"chorrusooc", Username:"<system>", UserID:"REDACTED", Avatar:"https://cdn.discordapp.com/avatars/REDACTED/a8cc07b5a83c3755d3c234bdd7fb5c0f.jpg", Account:"discord.alden-ic", Event:"", Protocol:"", Gateway:"", ParentID:"", Timestamp:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, ID:"REDACTED", Extra:map[string][]interface {}(nil)}
[0043] DEBUG gateway:      [SendMessage:gateway/gateway.go:498] mID discord.alden-ooc: REDACTED
[0043] DEBUG gateway:      [func1:gateway/gateway.go:488] => Send from discord.alden-ooc () to discord.alden-ooc (ID:REDACTED) took 105.182537ms
```

**Environment (please complete the following information):**
 - OS: Oracle Linux
 - Matterbridge version: 1.22.3 e45bbe45

**Additional context**
```toml
[irc.alden]
	Server="irc.sorcery.net:6667"
	Nick="PlanarBridge"
	Password="REDACTED"
	Label="Alden & Chorrus (IRC)"

[irc.safiralliance]
	Server="irc.darkmyst.org:6667"
	Nick="PlanarBridge"
	Password="REDACTED"
	Label="Safir Alliance (IRC)"

[discord.alden-ic]
	Token="REDACTED"
	Server="REDACTED-1"
	Label="Alden & Chorrus (Discord)"
	ShowJoinPart=false

[discord.alden-ooc]
	Token="REDACTED"
	Server="REDACTED-1"
	Label="Alden & Chorrus (Discord)"
	ShowJoinPart=true

[discord.safiralliance]
	Token="REDACTED"
	Server="REDACTED-2"
	Label="Safir Alliance (Discord)"
	ShowJoinPart=false

[general]
	RemoteNickFormat="<{NOPINGNICK}> "
	ShowEmbeds=false
	UseDiscriminator=false
	EditSuffix=" (edited)"
	ReplaceMessages=[ ["@here","`@here`"], ["@everyone","`@everyone`"] ]
	ShowTopicChange=true
	SyncTopic=true
	AllowMention=["roles", "users"]
	MessageClipped=" >"

[[gateway]]
	name = "alden"
	enable = true
	[[gateway.inout]]
		account = "discord.alden-ic"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.alden"
		channel = "#alden"

[[gateway]]
	name = "aldenooc"
	enable = true
	[[gateway.inout]]
		account = "discord.alden-ooc"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.alden"
		channel = "#aldenooc"

[[gateway]]
	name = "aldenplots"
	enable = true
	[[gateway.inout]]
		account = "discord.alden-ic"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.alden"
		channel = "#aldenplots"

[[gateway]]
	name = "aldenprivateroom"
	enable = true
	[[gateway.inout]]
		account = "discord.alden-ic"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.alden"
		channel = "#aldenprivateroom"

[[gateway]]
	name = "chorrusbar"
	enable = true
	[[gateway.inout]]
		account = "discord.alden-ic"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.alden"
		channel = "#chorrusbar"

[[gateway]]
	name = "chorrusooc"
	enable = true
	[[gateway.inout]]
		account = "discord.alden-ooc"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.alden"
		channel = "#chorrusooc"

[[gateway]]
	name = "chorrusplots"
	enable = true
	[[gateway.inout]]
		account = "discord.alden-ic"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.alden"
		channel = "#chorrusplots"

[[gateway]]
	name = "cshs"
	enable = true
	[[gateway.inout]]
		account = "discord.alden-ic"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.alden"
		channel = "#cshs"

[[gateway]]
	name = "zelathlir"
	enable = true
	[[gateway.inout]]
		account = "discord.safiralliance"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.safiralliance"
		channel = "#zelathlir"

[[gateway]]
	name = "lakuna"
	enable = true
	[[gateway.inout]]
		account = "discord.safiralliance"
		channel = "ID:REDACTED"
		[gateway.inout.options]
			WebhookURL="REDACTED"
	[[gateway.inout]]
		account = "irc.safiralliance"
		channel = "#lakuna"
```

**Comment by 42wim** [comment]
Are you talking about IRC joins that go to the 2 discord servers or are you talking about discord join from one server that gets to the other server?

Is it the same issue as #1507 ?

**Comment by 42wim** [comment]
Closing because no feedback

**Comment by wlcx** [comment]
I believe this bug is describing the below situation:
- A discord bot is present in multiple guilds (aka servers, using guild here for clarity)
- A matterbridge is configured with the bot's token and one of the guild's IDs, bridging to some other chat system
- A user joins/parts in one of the *other* guilds that the bot is present in
- matterbridge happily sends a join/part message because it doesn't check that the incoming join/part notification from discord is for the configured guild

**Comment by halian** [comment]
> I believe this bug is describing the below situation:
> 
> * A discord bot is present in multiple guilds (aka servers, using guild here for clarity)
> * A matterbridge is configured with the bot's token and one of the guild's IDs, bridging to some other chat system
> * A user joins/parts in one of the _other_ guilds that the bot is present in
> * matterbridge happily sends a join/part message because it doesn't check that the incoming join/part notification from discord is for the configured guild
> 
> Have fixed in #1846

Correct.
